### DISCUSSION TITLE
Store backup in GCS bucket

### DISCUSSION THREAD

**Discussion by Sagarbud** [nats-io/nats-server#5614] [General]
Hello All,

Is there any way to take full backup of NATS and store them into GCS bucket?

**Comment by atlejensen** [comment]
Not to my knowledge.
We have created a bash script that uses NATS cli to backup all accounts locally and then copy the files to a GCP bucket using gcloud cli.
We also backup the keys with nsc cli and gcloud cli.

**Comment by Sagarbud** [comment]
Hello @atlejensen 

Thank you for the update.

I will try and update here

**Comment by ripienaar** [comment]
`nats s backup` uses a library in jsm.go and since https://github.com/nats-io/jsm.go/pull/539 this library can write to any file handle.  I believe some users are using this to write their own backup/restore tooling that write to cloud buckets.
